include/pybind11/stl_bind.h

```diff
@@ -592,8 +603,26 @@ class_<Map, holder_type> bind_map(handle scope, const std::string &name, Args&&.
 
     cl.def(init<>());
 
+    cl.def(init([](const Map& d){
+            auto m = std::unique_ptr<Map>(new Map());
+            for (auto const &kv : d) {
+              m->insert(std::make_pair(kv.first, kv.second));
+            }
+            return m.release();
+    }));
+
+    cl.def(init([](const iterable& it) {
+          auto m = std::unique_ptr<Map>(new Map());
+            for (auto const& kv : pybind11::dict(it))
+                  m->insert(std::make_pair(kv.first.cast<KeyType>(), kv.second.cast<MappedType>()));
```

Comment:
Making a python dictionary and then iterating over that doesn't seem very efficient. We should directly get the keys/values from the iterable.

Reply:
@wjakob Can you please point me to something for this? Casting key and value fail here